WIP: For EmberTimes No. 103 - Unused Components and Explicit Service Injection (#161)
ember-learn/ember-blog: source/2019-06-21-the-ember-times-issue-103.md

<add your name to author list, top and bottom>
<add blurb and emoji to "SOME-INTRO-HERE">

---

## [Section Title in Title Case 🐹](#section-url)
<change section title emoji>
<consider adding some bold to your paragraph>

<add your name to author list, top and bottom>
<add blurb and emoji to "SOME-INTRO-HERE">

---

## [Talk Video: Visual Regression Testing Made Easy 📺](https://www.youtube.com/watch?v=d0ZCJNeybbs)

[@patocallaghan](https://github.com/patocallaghan) gave a lightning talk on “Visual Regression Testing Made Easy” at the April 2019 [DublinJS Meetup](https://www.meetup.com/DublinJS/). In his talk, Pat used [Percy](https://percy.io/) to scale testing visual regressions from manually doing them to having it automated. You can detect potential CSS bugs, remove certain types of tests, and make large refactoring and migrations. “Styles can sometimes make bad things happen.” Percy gives you confidence when you make big changes.

Check out the [full lightning talk video](https://www.youtube.com/watch?v=d0ZCJNeybbs) just published this week!

---

## [More Explicit Dependency Injection in Your Ember Apps 🐹](https://github.com/emberjs/rfcs/pull/502)

> The [**recent Request for Comments (RFC)**](https://github.com/emberjs/rfcs/pull/502), titled "Explicit Service Injection" proposes an update to the way **services and other dependencies** are registered, looked up and **injected** in Ember apps. The proposal argues that a class based syntax for [dependency injection](https://guides.emberjs.com/v3.10.0/applications/dependency-injection/) is more explicit than the current string based API and therefore preferable.

Reviewer: ```suggestion
The [**recent Request for Comments (RFC)**](https://github.com/emberjs/rfcs/pull/502), titled "Explicit Service Injection," proposes an update to the way **services and other dependencies** are registered, looked up and **injected** in Ember apps. The proposal argues that a class-based syntax for [dependency injection](https://guides.emberjs.com/v3.10.0/applications/dependency-injection/) is more explicit than the current string-based API and therefore preferable.
```

If you'd like to learn more about the **new API design** that is proposed in the RFC, be sure to [read the original proposal](https://github.com/nullvoxpopuli/rfcs/blob/explicit-dependency-injection/text/0000-explicit-dependency-injection.md) and do not forget to **join the discussion** in the comments!

---

## [Prune Your App with Ember Unused Components ♻️](https://github.com/vastec/ember-unused-components)

> Sorting out **components**, which are not used in your application anymore, can be tough. With different ways to invoke components, such as curly brace or [angle bracket syntax](https://guides.emberjs.com/release/reference/syntax-conversion-guide/) and the syntactical ambiguity of invoking helpers and components, it takes some time to prune your app from redundant components manually.

Reviewer: ```suggestion
Finding **components** that aren't used in your application anymore can be tough. Due to different ways to invoke components (with curly braces or [angle brackets](https://guides.emberjs.com/release/reference/syntax-conversion-guide/)) and the syntactical ambiguity between invoking helpers and components, it takes some time to prune your app from redundant components manually.
```
